app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt

```diff
@@ -55,7 +71,14 @@ internal class InitializationActivity : AppCompatActivity() {
 
         if (settings.tokenExists()) {
             runWithNeededPermissions {
-                tryAuthenticate()
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                    val manager = ContextCompat.getSystemService(this, AlarmManager::class.java)
+                    if (manager?.canScheduleExactAlarms() == false) {
+                        alarmDialog()
+                    } else {
+                        tryAuthenticate()
+                    }
+                }
```

Comment:
This looks a little suspicious, if the token exists and the SDK_INT is < VERSION_CODES.S, then nothing will be done, previously tryAuthenticate was called. Haven't tested this PR yet.

Reply:
True. That's exactly why I wanted to test it on other SDKs.  
I rather speeded the implementation :|